pkg/controller/labelidentity/label_group_index_test.go

```diff
@@ -0,0 +1,617 @@
+			expectMatch: false,
+		},
+		{
+			label:       labelC,
+			labelMatch:  labelMatchC,
+			selector:    selectorItemE,
+			expectMatch: false,
+		},
+	}
+	for _, tt := range tests {
+		matched := tt.labelMatch.matches(tt.selector)
+		if tt.expectMatch != matched {
+			t.Errorf("Unexpected matching status for %s and %s. Exp: %v, Act: %v", tt.label, tt.selector.getKey(), tt.expectMatch, matched)
+		}
+	}
+}
+
+func TestAddSelector(t *testing.T) {
+	tests := []struct {
+		name                 string
+		selectorToAdd        *types.GroupSelector
+		policyKey            string
+		expMatchedIDs        []uint32
+		expLabelIdentityKeys sets.String
+		expPolicyKeys        sets.String
+	}{
+		{
+			name:                 "cluster-wide app=web",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+			policyKey:            "policyA",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyA"),
+		},
+		{
+			name:                 "cluster-wide app=web another policy",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+			policyKey:            "policyB",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyA", "policyB"),
+		},
+		{
+			name:                 "pod app=web and ns purpose=test",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nsSelTest, nil, nil),
+			policyKey:            "policyB",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyB"),
+		},
+		{
+			name:                 "cluster-wide app=db",
+			selectorToAdd:        types.NewGroupSelector("", pSelDB, nil, nil, nil),
+			policyKey:            "policyC",
+			expMatchedIDs:        []uint32{2, 3},
+			expLabelIdentityKeys: sets.NewString(labelB, labelC),
+			expPolicyKeys:        sets.NewString("policyC"),
+		},
+		{
+			name:                 "app=db in ns testing",
+			selectorToAdd:        types.NewGroupSelector("testing", pSelDB, nil, nil, nil),
+			policyKey:            "policyD",
+			expMatchedIDs:        []uint32{2},
+			expLabelIdentityKeys: sets.NewString(labelB),
+			expPolicyKeys:        sets.NewString("policyD"),
+		},
+		{
+			name:                 "app=db in ns random",
+			selectorToAdd:        types.NewGroupSelector("random", pSelDB, nil, nil, nil),
+			policyKey:            "policyE",
+			expMatchedIDs:        []uint32{},
+			expLabelIdentityKeys: sets.NewString(),
+			expPolicyKeys:        sets.NewString("policyE"),
+		},
+	}
+	i := NewLabelIdentityIndex()
+	i.labelIdentities = map[string]*labelIdentityMatch{
+		labelA: labelMatchA,
+		labelB: labelMatchB,
+		labelC: labelMatchC,
+	}
+	i.labelIdentityNamespaceIndex = map[string]sets.String{
+		"testing": sets.NewString(labelA, labelB),
+		"nomatch": sets.NewString(labelC),
+	}
+	for _, tt := range tests {
```

Comment:
Missing the wrap line `t.Run(tt.name, func(t *testing.T) {}`, please check other test cases as well.

Reply:
Done